### PR TITLE
lib/config: Always set mtime window to 2 on Android unless ext detected (ref #7963)

### PR DESCRIPTION
**Pull request by tomasz1986**
The current detection is flawed, because it looks for a few specific
file systems like "msdos" or "fat" to set the mtime window, while in
reality Android seems to report names like "fuseblk", which can stand
for fat, ext4, or even f2fs.

At the moment, we set the mtime window only for a few known names used
for the fat filesystem. With this change, we take a safer approach of
always setting the time window unless we explicitly detect file systems
like ext2/ext3/ex4, which are known not to experience issues with moving
timestamps on Android.

### CODE REVIEW

**Comment by bt90** [comment]
What about f2fs?

**Comment by tomasz1986** [comment]
> What about f2fs?

I don't think it's detected at all. My systems report either nothing, or `fuseblk` on the `/sdcard` partition. It will have to suffer the 2s mtime window, unfortunately. It likely does already now though (when detected as `""`).

File: lib/config/folderconfiguration.go [hunk 1]
```diff
@@ -62,11 +62,11 @@ func (f FolderConfiguration) ModTimeWindow() time.Duration {
 		if usage, err := disk.Usage(f.Filesystem().URI()); err != nil {
 			dur = 2 * time.Second
 			l.Debugf(`Detecting FS at "%v" on android: Setting mtime window to 2s: err == "%v"`, f.Path, err)
-		} else if usage.Fstype == "" || strings.Contains(strings.ToLower(usage.Fstype), "fat") || strings.Contains(strings.ToLower(usage.Fstype), "msdos") {
+		} else if strings.Contains(strings.ToLower(usage.Fstype), "ext2") || strings.Contains(strings.ToLower(usage.Fstype), "ext3") || strings.Contains(strings.ToLower(usage.Fstype), "ext4") {
```
Comment (rasa) [inline lib/config/folderconfiguration.go:65]: How about flipping it around to:
`if strings.Contains("ext2ext3ext4", strings.ToLower(usage.Fstype)) {`
?

Comment (bt90) [reply lib/config/folderconfiguration.go:65]:
```
if strings.HasPrefix(strings.ToLower(usage.Fstype), "ext") {
```

Comment (tomasz1986) [reply lib/config/folderconfiguration.go:65]: > How about flipping it around to:
> `if strings.Contains("ext2ext3ext4", strings.ToLower(usage.Fstype)) {`
> ?

This won't work, will it? What we're actually looking for is likely `ext2/ext3` and/or possibly `ext2/ext3/ext4`.

> `if strings.HasPrefix(strings.ToLower(usage.Fstype), "ext") {`

This probably would be fine, but I personally think it's still safer to look for the values specifically. Who knows what Android will report there in the future 😕. Another possibility could be go regex and use something like `^ext[2-4]`, but I'm not sure if there's any real benefit over the current method.

**Review by calmh — APPROVED** [review]
I would have gone with just looking for the `ext` prefix but I don't really care strongly. 👍

**Pull request merged**